samples/snippets/quickstart.py

```diff
@@ -0,0 +1,269 @@
+#!/usr/bin/env python
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A sample script showing how to create, list and delete Google Compute Engine
+instances using the google-cloud-compute library. It can be run from command
+line to create, list and delete an instance in a given project in a given zone.
+"""
+
+# [START compute_instances_list]
+# [START compute_instances_list_all]
+# [START compute_instances_create]
+# [START compute_instances_delete]
+# [START compute_instances_operation_check]
+import typing
+
+import google.cloud.compute_v1 as compute_v1
+
+# [END compute_instances_operation_check]
+# [END compute_instances_delete]
+# [END compute_instances_create]
+# [END compute_instances_list_all]
+# [END compute_instances_list]
+
+
+# [START compute_instances_list]
+def list_instances(project_id: str, zone: str) -> typing.Iterable[compute_v1.Instance]:
+    """
+    Gets a list of instances created in given project in given zone.
+    Returns an iterable collection of Instance objects.
+
+    Args:
+        project_id: ID or number of the project you want to use.
+        zone: Name of the zone you want to check, for example: us-west3-b
+
+    Returns:
+        An iterable collection of Instance objects.
+    """
+    instance_client = compute_v1.InstancesClient()
+    instance_list = instance_client.list(project=project_id, zone=zone)
+
+    print(f"Instances found in zone {zone}:")
+    for instance in instance_list:
+        print(f" - {instance.name} ({instance.machine_type})")
+
+    return instance_list
+
+
+# [END compute_instances_list]
+
+
+# [START compute_instances_list_all]
```

Comment:
@m-strzelczyk thanks for the updated commit. There seems to be some mixup with compute_instances_create and compute_instances_list_all region tags.